src/components/SingleComment.jsx

```diff
@@ -5,9 +5,33 @@ import Chip from "@mui/joy/Chip";
 import Typography from "@mui/joy/Typography";
 import { convertTime } from "../utils/utils";
 import Divider from "@mui/joy/Divider";
+import { useContext, useState } from "react";
+import { UserContext } from "../context/UserContext";
+import { deleteSingleUserComment } from "../utils/api";
 
-const SingleComment = ({ comment }) => {
+const SingleComment = ({ comment, setComments }) => {
+  const [isError, setIsError] = useState(false);
+  const [isDeleted, setIsDeleted] = useState(false);
+  const { currentUser } = useContext(UserContext);
   const newTime = convertTime(comment.created_at);
+
+  const handleDelete = () => {
+    if (isDeleted) {
+      return;
+    }
+    setIsDeleted(true);
+    deleteSingleUserComment(comment.comment_id)
+      .then(() => {
+        setComments((currentComments) => {
+          return currentComments.slice(1);
```

Comment:
This appears to delete the first comment not the selected comment :(

Reply:
You will need to use a filter to remove the specific comment deleted.

Reply:
On it now! Not sure how I missed that...